MekHQ/src/mekhq/campaign/CampaignPreset.java

```diff
@@ -0,0 +1,496 @@
+        this.specialAbilities = specialAbilities;
+    }
+    //endregion Continuous
+    //endregion Getters/Setters
+
+    /**
+     * @return a list of all of the campaign presets in the default and userdata folders
+     */
+    public static List<CampaignPreset> getCampaignPresets() {
+        final List<CampaignPreset> presets = loadCampaignPresetsFromDirectory(
+                new File(MekHqConstants.CAMPAIGN_PRESET_DIRECTORY));
+        presets.addAll(loadCampaignPresetsFromDirectory(
+                new File(MekHqConstants.USER_CAMPAIGN_PRESET_DIRECTORY)));
+        final NaturalOrderComparator naturalOrderComparator = new NaturalOrderComparator();
+        presets.sort((p0, p1) -> naturalOrderComparator.compare(p0.toString(), p1.toString()));
+        return presets;
+    }
+
+    public void applyContinuousToCampaign(final Campaign campaign) {
+        if (getGameOptions() != null) {
+            campaign.setGameOptions(getGameOptions());
+            if (getCampaignOptions() == null) {
+                campaign.updateCampaignOptionsFromGameOptions();
+            }
+        }
+
+        if (getCampaignOptions() != null) {
+            campaign.setCampaignOptions(getCampaignOptions());
+            MekHQ.triggerEvent(new OptionsChangedEvent(campaign, getCampaignOptions()));
+        }
+
+        if (getRandomSkillPreferences() != null) {
+            campaign.setRandomSkillPreferences(getRandomSkillPreferences());
+        }
+
+        if (!getSkills().isEmpty()) {
+            SkillType.setSkillHash(getSkills());
+        }
+
+        if (!getSpecialAbilities().isEmpty()) {
+            SpecialAbility.setSpecialAbilities(getSpecialAbilities());
+        }
+    }
+
+    //region File I/O
+    public void writeToFile(final JFrame frame, @Nullable File file) {
+        if (file == null) {
+            return;
+        }
+        String path = file.getPath();
+        if (!path.endsWith(".xml")) {
+            path += ".xml";
+            file = new File(path);
+        }
+        try (OutputStream fos = new FileOutputStream(file);
+             OutputStream bos = new BufferedOutputStream(fos);
+             OutputStreamWriter osw = new OutputStreamWriter(bos, StandardCharsets.UTF_8);
+             PrintWriter pw = new PrintWriter(osw)) {
+            writeToXML(pw, 0);
+        } catch (Exception e) {
+            MekHQ.getLogger().error(e);
+            final ResourceBundle resources = ResourceBundle.getBundle("mekhq.resources.Campaign", new EncodeControl());
+            JOptionPane.showMessageDialog(frame, resources.getString("CampaignPresetSaveFailure.text"),
+                    resources.getString("CampaignPresetSaveFailure.title"), JOptionPane.ERROR_MESSAGE);
+        }
+    }
+
+    public void writeToXML(final PrintWriter pw, int indent) {
+        pw.println("<?xml version=\"1.0\" encoding=\"UTF-8\"?>");
+        MegaMekXmlUtil.writeSimpleXMLOpenTag(pw, indent++, "campaignPreset", "version", MekHQOptions.VERSION);
```

Comment:
I'm still kind of wondering why we're using this home-brewed serialization mechanism when you can use jaxb or some other third-party library that has already solved the problem. Some benefits include:
- we don't have to maintain a hand-rolled XML serialization library
- if we add a new property to the class, we don't have to update multiple different parsing/serialization routines

Reply:
Some negatives include increased difficulty in debugging and especially in migration, and how new fields are handled when added.

Also, we should never need to update multiple parsing routings for a single element, just the one.